src/main/msp/msp_serial.c

```diff
@@ -289,8 +291,10 @@ static int mspSerialSendFrame(mspPort_t *msp, const uint8_t * hdr, int hdrLen, c
     //     this allows us to transmit jumbo frames bigger than TX buffer (serialWriteBuf will block, but for jumbo frames we don't care)
     //  b) Response fits into TX buffer
     const int totalFrameLength = hdrLen + dataLen + crcLen;
+    const int perByteCostUs = (1000000 / (msp->port->baudRate / 10));
     if (!isSerialTransmitBufferEmpty(msp->port) && ((int)serialTxBytesFree(msp->port) < totalFrameLength)) {
-        return 0;
```

Comment:
This breaks current behavior of sending out a valid response. Maybe a directive can be used to isolate your change to only apply to maybe #define USE_OSD_OVER_MSP_DISPLAYPORT but I am not sure it's the right directive.